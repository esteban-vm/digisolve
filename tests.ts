import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
