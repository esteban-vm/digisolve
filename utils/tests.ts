import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
