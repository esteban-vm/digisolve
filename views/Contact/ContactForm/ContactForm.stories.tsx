import type { Meta } from '@storybook/react'
import { within, userEvent } from '@storybook/testing-library'
import { styled } from '@/styles'
import ContactForm from './ContactForm'

const Wrapper = styled.div`
  padding: 3rem 0;
  width: 65vw;
  background-color: var(--color-white);
`

export default {
  component: ContactForm,
  title: 'Views/Contact/Form',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const textField = canvas.getByPlaceholderText(/^your name$/i)
    const emailField = canvas.getByPlaceholderText(/^your email$/i)
    const messageArea = canvas.getByPlaceholderText(/^your message$/i)
    userEvent.type(textField, 'Test name')
    userEvent.type(emailField, 'test@example.com')
    userEvent.type(messageArea, 'Test message')
  },
} as Meta<typeof ContactForm>

export const FormExample = () => <ContactForm />
