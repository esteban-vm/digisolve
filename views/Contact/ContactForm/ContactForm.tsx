import type { BasicComponent } from '@/types'
import { Button } from '@/components'
import { styled } from '@/styles'

const ContactFormComponent: BasicComponent = (props) => {
  return (
    <form {...props} data-testid='form'>
      <div className='row'>
        <div className='col span_1_of_3'>
          <label htmlFor='name'>Name</label>
        </div>
        <div className='col span_2_of_3'>
          <input type='text' name='name' id='name' placeholder='Your name' required />
        </div>
      </div>

      <div className='row'>
        <div className='col span_1_of_3'>
          <label htmlFor='email'>Email</label>
        </div>
        <div className='col span_2_of_3'>
          <input type='email' name='email' id='email' placeholder='Your email' required />
        </div>
      </div>

      <div className='row'>
        <div className='col span_1_of_3'>
          <label htmlFor='newsletter'>Newsletter?</label>
        </div>
        <div className='col span_2_of_3'>
          <input type='checkbox' name='newsletter' id='newsletter' defaultChecked />
          Yes, please
        </div>
      </div>

      <div className='row'>
        <div className='col span_1_of_3'>
          <label htmlFor='message'>Drop us a line</label>
        </div>
        <div className='col span_2_of_3'>
          <textarea name='message' id='message' placeholder='Your message' />
        </div>
      </div>

      <div className='row'>
        <div className='col span_1_of_3' />
        <div className='col span_2_of_3'>
          <Button text='Send it!' isFull isSubmit />
        </div>
      </div>
    </form>
  )
}

const ContactForm = styled(ContactFormComponent)`
  text-align: left;
  width: 60%;
  margin: 0 auto;
  color: var(--color-grey);

  input[type='text'],
  input[type='email'],
  textarea {
    width: 100%;
    padding: 8px;
    border-radius: 3px;
    border: 1px solid #ccc;
    background-color: var(--color-white);
  }

  textarea {
    height: 100px;
    resize: none;
  }

  input[type='checkbox'] {
    margin: 10px 5px 10px 0;
  }

  label {
    user-select: none;
  }
`

export default ContactForm
