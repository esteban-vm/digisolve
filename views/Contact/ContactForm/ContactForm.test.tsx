import { render, cleanup, screen, create } from '@/tests'
import ContactForm from './ContactForm'

describe('ðŸ§ª CONTACT FORM:', () => {
  describe('display tests:', () => {
    let parent: HTMLElement

    beforeEach(() => {
      render(<ContactForm />)
      parent = screen.getByTestId('form')
    })

    afterEach(cleanup)

    it('should display a text field for name', () => {
      const field = screen.getByPlaceholderText(/^your name$/i)
      expect(field).toBeInTheDocument()
      expect(field).toBeVisible()
      expect(parent).toContainElement(field)
    })

    it('should display an email field', () => {
      const field = screen.getByPlaceholderText(/^your email$/i)
      expect(field).toBeInTheDocument()
      expect(field).toBeVisible()
      expect(parent).toContainElement(field)
    })

    it('should display a checkbox for newsletter', () => {
      const checkbox = screen.getByRole('checkbox', { name: /^newsletter\?$/i })
      expect(checkbox).toBeInTheDocument()
      expect(checkbox).toBeVisible()
      expect(parent).toContainElement(checkbox)
    })

    it('should display a text area for message', () => {
      const textArea = screen.getByPlaceholderText(/^your message$/i)
      expect(textArea).toBeInTheDocument()
      expect(textArea).toBeVisible()
      expect(parent).toContainElement(textArea)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<ContactForm />)
      expect(tree).toMatchSnapshot()
    })
  })
})
