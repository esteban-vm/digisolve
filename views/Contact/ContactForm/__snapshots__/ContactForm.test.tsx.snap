// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª CONTACT FORM: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: left;
  width: 60%;
  margin: 0 auto;
  color: var(--color-grey);
}

.emotion-0 input[type='text'],
.emotion-0 input[type='email'],
.emotion-0 textarea {
  width: 100%;
  padding: 8px;
  border-radius: 3px;
  border: 1px solid #ccc;
  background-color: var(--color-white);
}

.emotion-0 textarea {
  height: 100px;
  resize: none;
}

.emotion-0 input[type='checkbox'] {
  margin: 10px 5px 10px 0;
}

.emotion-0 label {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-1 {
  display: inline-block;
}

.emotion-1 a,
.emotion-1 input[type='submit'] {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-1 a:hover,
.emotion-1 input[type='submit']:hover,
.emotion-1 a:active,
.emotion-1 input[type='submit']:active {
  background-color: var(--color-dark-primary);
}

<form
  className="emotion-0"
  data-testid="form"
>
  <div
    className="row"
  >
    <div
      className="col span_1_of_3"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
    </div>
    <div
      className="col span_2_of_3"
    >
      <input
        id="name"
        name="name"
        placeholder="Your name"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col span_1_of_3"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
    </div>
    <div
      className="col span_2_of_3"
    >
      <input
        id="email"
        name="email"
        placeholder="Your email"
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col span_1_of_3"
    >
      <label
        htmlFor="newsletter"
      >
        Newsletter?
      </label>
    </div>
    <div
      className="col span_2_of_3"
    >
      <input
        defaultChecked={true}
        id="newsletter"
        name="newsletter"
        type="checkbox"
      />
      Yes, please
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col span_1_of_3"
    >
      <label
        htmlFor="message"
      >
        Drop us a line
      </label>
    </div>
    <div
      className="col span_2_of_3"
    >
      <textarea
        id="message"
        name="message"
        placeholder="Your message"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col span_1_of_3"
    />
    <div
      className="col span_2_of_3"
    >
      <span
        className="emotion-1"
      >
        <input
          type="submit"
          value="Send it!"
        />
      </span>
    </div>
  </div>
</form>
`;
